hbase-handler/src/java/org/apache/hadoop/hive/hbase/HBaseStorageHandler.java

```diff
@@ -296,7 +297,8 @@ public URI getURIForAuth(Table table) throws URISyntaxException {
     String hbase_host = tableProperties.containsKey(HBASE_HOST_NAME)? tableProperties.get(HBASE_HOST_NAME) : hbaseConf.get(HBASE_HOST_NAME);
     String hbase_port = tableProperties.containsKey(HBASE_CLIENT_PORT)? tableProperties.get(HBASE_CLIENT_PORT) : hbaseConf.get(HBASE_CLIENT_PORT);
     String table_name = tableProperties.getOrDefault(HBaseSerDe.HBASE_TABLE_NAME, null);
```

Comment:
Is the table name guaranteed not to have symbols which require URL-encoding?

Reply:
So I was thinking about doing it for the whole URI and not just the properties.
I didn't research the characters that get URL Encoded versus the characters that are used in a table, but I was a bit nervous to URL Encode the table name which might break backward compatibility.  

I figured URL encoding Serde properties was pretty safe though since it probably isn't used by Ranger.  I also thought about removing Serde properties from the URI, but that also had me worried about backward compatibility.

I'm open to changing this or having more of a discussion on it.

Reply:
Changed it so we're URLEncoding the whole URI now.